src/components/SitePreview.js

```diff
@@ -0,0 +1,151 @@
+// 3rd party libraries - core
```

Comment:
Nearly all the code below is exactly what used to be in the `SearchMap` screen, but it's now refactored into its own component.